### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by DijonnaWatson**
added one of the tests using the prototype for FactorComputationCordinator API. Also trying to check the gradle and see if doing a pull request runs the gradle.

### CODE REVIEW

**Review by ViceRaven — APPROVED** [review]
The code looks good. I went over step 7 to make sure we have everything.

**Comment by DijonnaWatson** [comment]
everything looks good for the assignment so whoever wants to approve and merge can do it.

File: test/InMemoryInputConfig.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  public InMemoryInputConfig(List<Integer> inputData) {
+    this.inputData = new ArrayList<>(inputData);
+  }
+
+  @Override
+  public List<Integer> getInputData() {
+    return new ArrayList<>(inputData);
+  }
+}
```
Comment (jeremy23newman) [inline test/InMemoryInputConfig.java:14]: looks good to me, fits the specifications

File: test/InMemoryOutputConfig.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  @Override
+  public void writeOutputData(List<String> data) {
+    outputData.addAll(data);
+  }
+
+  @Override
+  public List<String> getOutputData() {
+    return new ArrayList<>(outputData);
+  }
+}
```
Comment (jeremy23newman) [inline test/InMemoryOutputConfig.java:19]: looks good to me

File: test/TestComputeEngineIntegration.java
Comment (DijonnaWatson) [inline test/TestComputeEngineIntegration.java:1]: This looks good Jeremy

File: test/TestFactorComputeEngine.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    // coupled as far as the implementation (how to format a result really
+    // depends on what that result represents), this is reasonable to put in one
+    // component, but keeping things more general is also fine. API Design is
+    // much more art than science!
+    int[] result = engine.getFactors(10); // Testing the mockEngine
+
+    // The mock worked since it printed the correct the array of factors
+    System.out.println("Result: " + java.util.Arrays.toString(result));
+  }
+}
```
Comment (jeremy23newman) [inline test/TestFactorComputeEngine.java:41]: good comments on the new additions to the class

File: test/TestFactorDataStore.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  public int[] testretrieveFactors(int i) {
+    // TODO Auto-generated method stub
+    return null;
+  }
+  // temporary method in order to test the compute engine integration
+  // stores factors from number to be checked
+  public void teststoreFactors(int number, int[] factors) {
+    // TODO Auto-generated method stub
+  }
+}
```
Comment (jeremy23newman) [inline test/TestFactorDataStore.java:116]: i think this is good, good job

**Review by jeremy23newman — APPROVED** [review]
overall well done guys

**Review by ViceRaven — COMMENTED** [review]
The in-memory java files had a few issues. I had to fix the formatting and add jeremy's package. Also for the InMemoryFactoryDataStore file I had accidentally added the write result class in there which was making the checkstyle fail. After removing it the InMemoryFactoryDataStore  file passed the checkstyle.

**Comment by DijonnaWatson** [comment]
> The code looks good. I went over step 7 to make sure we have everything.

same

File: test/InMemoryFactorDataStore.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.example.factorfinder;
+import com.example.factorfinder.WriteResult;
```
Comment (ViceRaven) [inline test/InMemoryFactorDataStore.java:2]: This import statement was what was missing. The checkstyle passes now.

File: test/InMemoryFactorDataStore.java
Comment (DijonnaWatson) [inline test/InMemoryFactorDataStore.java:1]: This looks great.

**Pull request merged**